client/shared/src/api/extension/flatExtensionApi.ts

```diff
@@ -226,6 +228,13 @@ export const initNewExtensionAPI = (
         provider: sourcegraph.DefinitionProvider
     ): sourcegraph.Unsubscribable => addWithRollback(state.definitionProviders, { selector, provider })
 
+    // GraphQL
+    const graphQL: typeof sourcegraph['graphQL'] = {
+        requestGraphQL: <TResult, TVariables extends object>(query: string, variables: TVariables) =>
+            // Remote functions lose generic type parameters, so we have to cast the result
+            (mainAPI.requestGraphQL(query, variables) as unknown) as Promise<GraphQLResult<TResult>>,
```

Comment:
This will work if above suggestion is applied since `any` will be assignable and the rest is inferred
```suggestion
        requestGraphQL: (query, variables) => mainAPI.requestGraphQL(query, variables)
```